Update of TPC fast space charge correction with splines

davidrohr opened the pull request:
Rebased version of #14723

sgorbuno committed:
TPC Splines: fix initialization of the track residuals in the test macro

sgorbuno committed:
TPC Splines: fix propagation of the track residual data to the TPC row edges

sgorbuno committed:
TPC Splines: non-uniform grid that corresponds to the track residual voxels

sgorbuno committed:
TPC Splines: multithreaded reading of the residual tree

sgorbuno committed:
TPC Splines: add limits for SP correction values per TPC row

sgorbuno committed:
TPC Splines: disable smoothing

Commit:
TPC Splines: smooth to linear edges, crop at grid borders, use mean position of residuals

sgorbuno committed:
TPC Splines: fix the inverse correction

sgorbuno committed:
TPC Splines: fix reading track residuals

sgorbuno committed:
TPC Splines: fix scaling splines outside of the measured area

sgorbuno committed:
TPC Splines: rename Slice -> Roc in geometry

sgorbuno committed:
TPC Splines: minimise the amount of transformations

sgorbuno committed:
TPC Splines: init inverse from the inverse voxel map; rebase

sgorbuno committed:
TPC Splines: cleanup

sgorbuno committed:
TPC Splines: fast merge of SC corrections

sgorbuno committed:
TPC Splines: get rid of internal UV coordinates

sgorbuno committed:
TPC Splines: completely switch to local TPC coordinates in the grid

cbmsw committed:
TPC Splines: correct biased voxels; features for debugging

cbmsw committed:
TPC Splines: replace std::tuple by std::array

cbmsw committed:
TPC Splines: better smoothing between the voxels

davidrohr committed:
TPCFastTransform: fix compilation on GPU with the new splines

cbmsw committed:
TPC Splines: bugfixes in spline merging

cbmsw committed:
TPC Splines: add backward compatibility

cbmsw committed:
Fix compiler-warnings, codechecker violations and compilation

Fix compiler warnings on MacOS

Fix compiler warning, memmove must only operate on trivial types

Fix coding rule violations

TPC Splines: compilation fix

matthias-kleiner committed:
TPC: centralize correction map building in TPCScalerSpec

`TPCScalerSpec` now owns the full correction map pipeline: loads maps from CCDB, applies luminosity scaling and M-shape corrections, applies VDrift into the final map, and publishes a single merged `TPCFastTransformPOD` (`TPC/TPCCORRMAP`) plus instantaneous CTP lumi (`CTP/LUMICTP`) each TF.

Downstream consumers (ITS-TPC matcher, TRD, TOF, secondary vertexing, GPU reco, etc.) are simplified: lumi scale options removed from constructors, `corrMapsLoader.updateVDrift()` / `corrMapsLoader.accountCCDBInputs()` / `corrMapsLoader.init()` calls dropped, and `requestCCDBInputs()` replaced by `requestInputs()` subscribing only to the two new outputs.

`TPCScalerSpec` is now always added unconditionally.

`CorrectionMapsHelper` is reduced to a thin wrapper around `TPCFastTransformPOD*`; full lumi-scaling state moves to the new `CorrectionMapsHelperFull` / `CorrectionMapsLoaderFull` used only inside `TPCScalerSpec`. `TPCFastTransformPOD` gains a flat buffer offset table and `interpolateAtUZeroCopy()` for correct spline evaluation on zero-copy shared memory paths.

alibuild commented:
Error while checking build/O2/fullCI_slc9 for dc97a9dbfb71e41b321757e02250bcb37ba30884 at 2025-12-15 11:27:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/14921-slc9_x86-64/0/GPU/GPUTracking/Merger/GPUTPCGMTrackParam.cxx:525:270: error: no matching function for call to 'o2::gpu::TPCFastTransform::convVertexTimeToZOffset(int32_t&, float&, const int32_t&) const'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/dc97a9dbfb71e41b321757e02250bcb37ba30884/build_O2_fullCI_slc9/pretty.html).

davidrohr commented:
Dataflow issue is independent, and compiles and runs on GPU, but should not be merged yet. Will comment more in the mail thread.

cbmsw committed:
TPC Splines: keep old cropping scheme

alibuild commented:
Error while checking build/O2/fullCI_slc9 for d30e01aef237ebb6bb56f696a6c4b01857d6bc26 at 2026-03-07 23:36:
```
## sw/BUILD/O2Physics-latest/log
c++: fatal error: Killed signal terminated program cc1plus
c++: fatal error: Killed signal terminated program cc1plus
c++: fatal error: Killed signal terminated program cc1plus
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/d30e01aef237ebb6bb56f696a6c4b01857d6bc26/build_O2_fullCI_slc9/pretty.html).

shahor02 committed:
Add POD version of TPCFastTransform

The TPCFastTransformPOD is a pointerless version of the TPCFastTransform. It can be created from the
original TPCFastTransform as e.g.
auto lold = o2::gpu::TPCFastTransform::loadFromFile("o2-gpu-TPCFastTransform.root","ccdb_object"); // load original transform
std::vector<char> v; // one has to provide a vector (could be a std or pmr), which later can be messaged via DPL
auto* pod = o2::gpu::TPCFastTransformPOD::create(v, *lold); // pointer pod is just v.data() cast to TPCFastTransformPOD*

// run test:
pod->test(*lold);
[INFO]  (ns per call)              original        this     Nmissmatch
[INFO] getCorrection               1.330e+02    1.400e+02   0
[INFO] getCorrectionInvCorrectedX  8.856e+01    8.434e+01   0
[INFO] getCorrectionInvUV          6.266e+01    6.142e+01   0

It can be also created directly from the TPCFastSpaceChargeCorrection as
TPCFastSpaceChargeCorrection& oldCorr = lold->getCorrection();
auto* pod = o2::gpu::TPCFastTransformPOD::create(v, oldCorr);
but in this case one should afterwards set the vdrift and t0 using provided getters.

TPCFastTransformPOD replicates all the methods of the TPCFastTransform (and of the TPCFastSpaceChargeCorrection), including
those which allow to query rescaled corrections (by providing refernce maps and scaling coefficients).
Since the idea of this class is to create a final correction map as a weighted sum of different contribution and to distribute
it to consumer processes via shared memory, also the query methods w/o rescaling are added, they have the suffix _new added.
Eventually, the scalable legacy methods can be suppressed and the suffix new can be dropped.

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 6a7d6f6da882429b4f0748f5a4e7b0bb61139f29 at 2026-03-26 20:06:
```
## sw/BUILD/O2Physics-latest/log
c++: fatal error: Killed signal terminated program cc1plus
c++: fatal error: Killed signal terminated program cc1plus
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/6a7d6f6da882429b4f0748f5a4e7b0bb61139f29/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
GPU TPC FastTransformation: Do not use std::array

davidrohr committed:
GPU TPCFastTransformation: Do not use double

davidrohr committed:
GPU TPCFastTransfomration: Add missing inline keywords

matthias-kleiner committed:
TPC: remove CorrectionMaps helper/loader classes

Replace CorrectionMapsLoader/Helper with their Full variants,
renaming CorrectionMapsLoaderFull -> CorrectionMapsLoader and
CorrectionMapsHelperFull -> CorrectionMapsHelper.
Use TPCFastTransformPOD directly without the intermediate helper class.

davidrohr committed:
GPU FlatObject: Make some more functions accessible on GPU

davidrohr committed:
GPU TPCTransform: new POD transform does not need support for applying multiple maps at once, simplify and remove _new functions

davidrohr committed:
GPU: Add GPUCommonAlignedAlloc.h for aligned buffers, and use it for TPCFastTransformPOD

davidrohr committed:
GPU Standalone: Add dumping and reading of dynamic structs with larger buffer than sizeof(struct)

alibuild commented:
Error while checking build/O2/fullCI_slc9 for bfe3084c0ba3449ad0b70d84b0fef17e5873e76e at 2026-04-04 04:24:
```
## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc9_x86-64/O2/slc9_x86-64-slc9_x86-64-local10/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 1
[ERROR] invalid workflow in o2-tpc-scaler-workflow: No matching output found for CTP/LUMI/0 as requested by data processor "tpc-scaler". Candidates:
[ERROR] -DPL/ENUM/1301969186
[ERROR] -CPV/RAWDATA
[ERROR] -CTP/RAWDATA
[ERROR] -EMC/RAWDATA
[ERROR] -FDD/RAWDATA
[ERROR] -FLP/DISTSUBTIMEFRAME/0
[ERROR] -FLP/DISTSUBTIMEFRAME/52443
[ERROR] -FT0/RAWDATA
[ERROR] -FV0/RAWDATA
[ERROR] -HMP/RAWDATA
[ERROR] -ITS/RAWDATA
[ERROR] -MCH/RAWDATA
[ERROR] -MFT/RAWDATA
[ERROR] -MID/RAWDATA
[ERROR] -PHS/RAWDATA
[ERROR] -TOF/RAWDATA
[ERROR] -TPC/RAWDATA
[ERROR] -TRD/RAWDATA
[ERROR] -ZDC/RAWDATA
[ERROR] -CTP/ORBITRESET/0
[ERROR] -CTP/OrbitReset/0
[ERROR] -TPC/CalibLaserTracks/0
[ERROR] -TPC/CorrMap/0
[ERROR] -TPC/CorrMapParam/0
[ERROR] -TPC/CorrMapRef/0
[ERROR] -TPC/PRESSURECCDB/0
[ERROR] -TPC/TEMPERATURECCDB/0
[ERROR] -TPC/VDriftTgl/0
[ERROR] -CTP/LUMICTP/0
[ERROR] -TPC/TPCCORRMAP/0
[ERROR] 


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
grep: error-log.txt: binary file matches
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/77d37cf60bc85bced9c62e1e4a4f0080cf0a9e41/slc9_x86-64/o2checkcode/1.0-local36/etc/modulefiles
++ cat
--

[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/bfe3084c0ba3449ad0b70d84b0fef17e5873e76e/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for ae25f0e056edddc60eb955501ea58fa281fa43f6 at 2026-04-04 13:26:
```
## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc9_x86-64/O2/slc9_x86-64-slc9_x86-64-local10/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 1
[ERROR] invalid workflow in o2-tpc-scaler-workflow: No matching output found for CTP/LUMI/0 as requested by data processor "tpc-scaler". Candidates:
[ERROR] -DPL/ENUM/1301969186
[ERROR] -CPV/RAWDATA
[ERROR] -CTP/RAWDATA
[ERROR] -EMC/RAWDATA
[ERROR] -FDD/RAWDATA
[ERROR] -FLP/DISTSUBTIMEFRAME/0
[ERROR] -FLP/DISTSUBTIMEFRAME/52443
[ERROR] -FT0/RAWDATA
[ERROR] -FV0/RAWDATA
[ERROR] -HMP/RAWDATA
[ERROR] -ITS/RAWDATA
[ERROR] -MCH/RAWDATA
[ERROR] -MFT/RAWDATA
[ERROR] -MID/RAWDATA
[ERROR] -PHS/RAWDATA
[ERROR] -TOF/RAWDATA
[ERROR] -TPC/RAWDATA
[ERROR] -TRD/RAWDATA
[ERROR] -ZDC/RAWDATA
[ERROR] -CTP/ORBITRESET/0
[ERROR] -CTP/OrbitReset/0
[ERROR] -TPC/CalibLaserTracks/0
[ERROR] -TPC/CorrMap/0
[ERROR] -TPC/CorrMapParam/0
[ERROR] -TPC/CorrMapRef/0
[ERROR] -TPC/PRESSURECCDB/0
[ERROR] -TPC/TEMPERATURECCDB/0
[ERROR] -TPC/VDriftTgl/0
[ERROR] -CTP/LUMICTP/0
[ERROR] -TPC/TPCCORRMAP/0
[ERROR] 


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
grep: error-log.txt: binary file matches
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/32d92becb33395cf7d7d710221b8da4eba3f7cbc/slc9_x86-64/o2checkcode/1.0-local39/etc/modulefiles
++ cat
--

[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/ae25f0e056edddc60eb955501ea58fa281fa43f6/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 1fac169a3f384bec98aed82abda880442839d5c0 at 2026-04-07 13:04:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/1fac169a3f384bec98aed82abda880442839d5c0/build_O2_fullCI_slc9/pretty.html).

matthias-kleiner committed:
make TPCFastTransformPOD fully independent of FlatObject and root

davidrohr committed:
GPU OpenGL Display: remove broken functionality to draw text via ImGui

davidrohr committed:
Revert "GPU Display: temporarily disable imgui text rendering, for imgui bump"

This reverts commit 05498e8dcae160b9be89f8170f693325c6ccd6ee.

davidrohr committed:
GPU: Renaming and removal of obsolete macros / defines

davidrohr committed:
GPU: Change some defines to constexpr variables, some renaming

davidrohr committed:
GPU: Remove obsolete file

davidrohr committed:
GPU: Unify some constants

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 6097901fc2ba3af7113e65ea042a6109c2f5511f at 2026-04-08 11:16:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/6097901fc2ba3af7113e65ea042a6109c2f5511f/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
Fix compiler warning

davidrohr committed:
GPU: Fix compilation in Run2 mode

matthias-kleiner committed:
removing remaining pointers and classdef from TPCFastTransformPOD

davidrohr committed:
GPU: Add ctest for GPUTPCGeometry

davidrohr committed:
TPC: Make TPC calibration objects constructible from standalone benchmark

davidrohr committed:
TPC FastTransform: Get Geometry from GPUTPCGeometry not from o2::tpc::mapper

davidrohr committed:
GPU: Add workaround if fmt not available to GPUCommonLogger

matthias-kleiner committed:
handle disable-ctp-lumi-request properly in all workflows

davidrohr committed:
GPU Common: Add GPUCommonConfigurableParam to support ConfigurableParam classes in GPU code

davidrohr committed:
GPU Standalone: Add recreateTrivialCalibObjects option

davidrohr committed:
Remove unnecessary GPUCA_GPUCODE protection in non-GPU code

davidrohr committed:
GPU: Add check that struct was read correctly from file

alibuild commented:
Error while checking build/O2/fullCI_slc9 for db8aab4c0619d8efe063a2a3decceffd55667232 at 2026-04-09 19:07:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
grep: error-log.txt: binary file matches
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/db8aab4c0619d8efe063a2a3decceffd55667232/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
GPU TPCFastTransform: Use constexpr GPUTPCGeometry where possible

davidrohr committed:
GPU: Add unfolded padWidthPerRow to GPUTPCGeometry

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 61e8d8678b26d6edf7c8f448997c74d42c46aa5b at 2026-04-10 14:39:
```
## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc9_x86-64/O2/slc9_x86-64-slc9_x86-64-local11/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 1
[ERROR] invalid workflow in o2-tpc-scaler-workflow: No matching output found for CTP/LUMI/0 as requested by data processor "tpc-scaler". Candidates:
[ERROR] -DPL/ENUM/1301969186
[ERROR] -CPV/RAWDATA
[ERROR] -CTP/RAWDATA
[ERROR] -EMC/RAWDATA
[ERROR] -FDD/RAWDATA
[ERROR] -FLP/DISTSUBTIMEFRAME/0
[ERROR] -FLP/DISTSUBTIMEFRAME/52443
[ERROR] -FT0/RAWDATA
[ERROR] -FV0/RAWDATA
[ERROR] -HMP/RAWDATA
[ERROR] -ITS/RAWDATA
[ERROR] -MCH/RAWDATA
[ERROR] -MFT/RAWDATA
[ERROR] -MID/RAWDATA
[ERROR] -PHS/RAWDATA
[ERROR] -TOF/RAWDATA
[ERROR] -TPC/RAWDATA
[ERROR] -TRD/RAWDATA
[ERROR] -ZDC/RAWDATA
[ERROR] -CTP/ORBITRESET/0
[ERROR] -CTP/OrbitReset/0
[ERROR] -TPC/CalibLaserTracks/0
[ERROR] -TPC/CorrMap/0
[ERROR] -TPC/CorrMapParam/0
[ERROR] -TPC/CorrMapRef/0
[ERROR] -TPC/PRESSURECCDB/0
[ERROR] -TPC/TEMPERATURECCDB/0
[ERROR] -TPC/VDriftTgl/0
[ERROR] -CTP/LUMICTP/0
[ERROR] -TPC/TPCCORRMAP/0
[ERROR] 


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
grep: error-log.txt: binary file matches
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/35b134d3a9d63dc5e896e57b0e12960152dac121/slc9_x86-64/o2checkcode/1.0-local56/etc/modulefiles
++ cat
--

[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/61e8d8678b26d6edf7c8f448997c74d42c46aa5b/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
GPU: Add sector angles to GPUTPCGeometry

davidrohr committed:
GPU TRD: Reduce headers included in interface class

davidrohr committed:
GPU: Move some files such that GPUDataTypes does not depend on TPCFastTransformation

alibuild commented:
Error while checking build/O2/fullCI_slc9 for e20811206ed9ec123f736d69dcbba748449eac9e at 2026-04-10 23:02:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/e20811206ed9ec123f736d69dcbba748449eac9e/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
GPU: Add tuned parameters for NVIDIA Blackwell

alibuild commented:
Error while checking build/O2/fullCI_slc9 for a84a88c4a1eb056b6a1856bd5ec2672fcbf07857 at 2026-04-13 15:53:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/a84a88c4a1eb056b6a1856bd5ec2672fcbf07857/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
GPU: Add GPUCA_DETERMINISTIC_NO_FTC

davidrohr committed:
GPU: GPULooger should suppress messages below info level in standalone benchmark

davidrohr committed:
GPU: Write floats in hex format to debug dump when in deterministic mode

davidrohr committed:
GPU: Restructure parameters for TPC track merging, simplifying and working around an NVIDIA compiler bug for blackwell

davidrohr committed:
GPU: Fix compiler warning

davidrohr committed:
GPU TPC Clusterizer: Add static assert to forbit invalid thread counts

davidrohr committed:
GPU Standalone: Fix TPC_MAX_TIME_BIN_TRIGGERED for Run2 data

davidrohr committed:
GPU: Use GPUCommonAlignedAlloc.h in more places

davidrohr committed:
GPU Standalone: Fix corner cases when no TRD data present, TRDRecoParam missing, and address sanitizer warnings

davidrohr committed:
TPCFastTransform: Do not pollute the global namespace

alibuild commented:
Error while checking build/O2/fullCI_slc9 for f5de2b03954f716c111a2d49db7625b43e7cb53a at 2026-04-13 20:34:
```
## sw/BUILD/O2-latest/log
ninja: error: dependency cycle: GPU/GPUTracking/G__O2GPUDataTypes.cxx -> stage/lib64/G__O2TPCFastTransformation_rdict.pcm -> stage/lib64/libO2GPUDataTypes.so -> GPU/GPUTracking/CMakeFiles/O2lib-GPUDataTypes.dir/DataTypes/GPUDataTypesConfig.cxx.o -> cmake_object_order_depends_target_O2lib-GPUDataTypes -> GPU/GPUTracking/G__O2GPUDataTypes.cxx
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/f5de2b03954f716c111a2d49db7625b43e7cb53a/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 3b23e9a1471fc2b74f0f58e53af46b597ff7e6ed at 2026-04-14 05:39:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
grep: error-log.txt: binary file matches
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/3b23e9a1471fc2b74f0f58e53af46b597ff7e6ed/build_O2_fullCI_slc9/pretty.html).

davidrohr committed:
GPU CMake: Simplify header install logic for GPUCommon

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 3410b5d29957f41576a6b1b32cd7491e2d640aa4 at 2026-04-14 09:41:
```
## sw/BUILD/O2-latest/log
ninja: error: dependency cycle: GPU/GPUTracking/G__O2GPUDataTypes.cxx -> stage/lib64/G__O2TPCFastTransformation_rdict.pcm -> stage/lib64/libO2GPUDataTypes.so -> GPU/GPUTracking/CMakeFiles/O2lib-GPUDataTypes.dir/DataTypes/GPUDataTypesConfig.cxx.o -> cmake_object_order_depends_target_O2lib-GPUDataTypes -> GPU/GPUTracking/G__O2GPUDataTypes.cxx
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/3410b5d29957f41576a6b1b32cd7491e2d640aa4/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for e8874be49a286f8c4e80f9ee62a8bad282c2b62f at 2026-04-14 14:25:
```
## sw/BUILD/O2-latest/log
ninja: error: dependency cycle: GPU/GPUTracking/G__O2GPUDataTypes.cxx -> stage/lib64/G__O2TPCFastTransformation_rdict.pcm -> stage/lib64/libO2GPUDataTypes.so -> GPU/GPUTracking/CMakeFiles/O2lib-GPUDataTypes.dir/DataTypes/GPUDataTypesConfig.cxx.o -> cmake_object_order_depends_target_O2lib-GPUDataTypes -> GPU/GPUTracking/G__O2GPUDataTypes.cxx
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/e8874be49a286f8c4e80f9ee62a8bad282c2b62f/build_O2_fullCI_slc9/pretty.html).

cbmsw committed:
TPCFastTransform: adjust the test macro to the new code and add test for POD version

davidrohr committed:
GPU: Improve debug / info messages

cbmsw committed:
TPCFastTransform: remove row-wise max correction values

davidrohr committed:
GPU: Add python script to convert C++ parameter header to csv file

davidrohr commented:
Converting it to draft, since with the current incremental build issue, this makes other PRs fail, when they try to build incrementally based on this one.

davidrohr committed:
GPU CMake: Fix should fail when no parameters for architecture exist

davidrohr committed:
GPU: Enforce some GPU-kernels are RTC-compiled with correct WARP_SIZE as launch-bounds

cbmsw committed:
TPCFastTransform: mirror Z-´of the spline grid for the C side

davidrohr committed:
GPU TPC Merging: Fix alternateBorderSort=0 with GPU in per-kernel mode

davidrohr committed:
GPU: Change some default GPU parameters

cbmsw committed:
TPCFastTransform: reduce the metadata: the same spline setup for all the sectors

cbmsw committed:
TPCFastTransform: renaming

cbmsw committed:
TPCFastTransform: remove unnecessary memory alignment in the spline data

davidrohr committed:
GPU OpenCL: Fix support for half precision

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 03ca14767e002137ad48e83fa962b4c08dcda823 at 2026-04-18 17:41:
```
## sw/BUILD/O2-latest/log
ninja: error: dependency cycle: stage/lib64/G__O2GPUDataTypes_rdict.pcm -> stage/lib64/libO2TPCFastTransformation.so -> GPU/TPCFastTransformation/CMakeFiles/O2lib-TPCFastTransformation.dir/SplineSpec.cxx.o -> cmake_object_order_depends_target_O2lib-TPCFastTransformation -> GPU/TPCFastTransformation/G__O2TPCFastTransformation.cxx -> stage/lib64/G__O2GPUDataTypes_rdict.pcm


## sw/BUILD/QualityControl-latest/log
CMake Error at CMakeLists.txt:99 (find_package):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/03ca14767e002137ad48e83fa962b4c08dcda823/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for c1ea6a9c15b0569e2885421a4cbbd334b68a27fa at 2026-04-19 07:47:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Common/DCAFitter/GPU/cuda/GPUInterface.cu:55:15: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/tracking/src/LineVertexerHelpers.cxx:746:65: error: prefer transparent functors 'greater<>' [modernize-use-transparent-functors]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/c1ea6a9c15b0569e2885421a4cbbd334b68a27fa/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 7172ab0b15f367bdaa4e69eed900369e6502b602 at 2026-04-19 18:28:
```
## sw/BUILD/O2-latest/log
ninja: error: dependency cycle: GPU/GPUTracking/G__O2GPUDataTypes.cxx -> stage/lib64/G__O2TPCFastTransformation_rdict.pcm -> stage/lib64/libO2GPUDataTypes.so -> GPU/GPUTracking/CMakeFiles/O2lib-GPUDataTypes.dir/DataTypes/GPUDataTypesConfig.cxx.o -> cmake_object_order_depends_target_O2lib-GPUDataTypes -> GPU/GPUTracking/G__O2GPUDataTypes.cxx
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/7172ab0b15f367bdaa4e69eed900369e6502b602/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 8bcee51992fee482c5670859fb4c90e408705945 at 2026-04-21 08:51:
```
## sw/BUILD/O2-latest/log
ninja: error: dependency cycle: GPU/GPUTracking/G__O2GPUDataTypes.cxx -> stage/lib64/G__O2TPCFastTransformation_rdict.pcm -> stage/lib64/libO2GPUDataTypes.so -> GPU/GPUTracking/CMakeFiles/O2lib-GPUDataTypes.dir/DataTypes/GPUDataTypesConfig.cxx.o -> cmake_object_order_depends_target_O2lib-GPUDataTypes -> GPU/GPUTracking/G__O2GPUDataTypes.cxx
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14921/8bcee51992fee482c5670859fb4c90e408705945/build_O2_fullCI_slc9/pretty.html).

shahor02 commented on Detectors/Align/Workflow/src/BarrelAlignmentSpec.cxx:
```diff
@@ -122,7 +117,9 @@ class BarrelAlignmentSpec : public Task
   o2::tpc::VDriftHelper mTPCVDriftHelper{};
-  o2::tpc::CorrectionMapsLoader mTPCCorrMapsLoader{};
+
+  o2::gpu::aligned_unique_buffer_ptr<o2::gpu::TPCFastTransformPOD> mCorrMapBuffer; // buffer to hold the raw map data from CCDB, needed to keep the pointer valid in the CorrectionMapsHelper
```
What is this buffer for?

davidrohr replied:
indeed, it is not used. Removed it.

shahor02 reviewed (dismissed):
Hi, looks fine for me, just a small comment below (probably to @matthias-kleiner ) about the unused variable.
